homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,253 @@
+        """Return True if entity is on."""
+        if self._current_mode is None:
+            return None
+        return cast(bool, int(self._current_mode.value) != HumidityControlMode.OFF)
+
+    @property
+    def target_humidity(self) -> int | None:
+        """Return the humidity we try to reach."""
+        if not self.is_on or self._setpoint_value is None:
+            return None
+        return int(self._setpoint_value.value)
+
+    @property
+    def mode(self) -> str | None:
+        """Return the current humidity control mode."""
+        if self._current_mode:
+            return cast(
+                str, self._current_mode.metadata.states[str(self._current_mode.value)]
+            )
+        return None
+
+    async def async_set_mode(self, mode: str) -> None:
+        """Set new mode."""
+
+        if self._current_mode is None:
+            return None
+
+        try:
+            new_state = int(
+                next(
+                    state
+                    for state, label in self._current_mode.metadata.states.items()
+                    if label == mode
+                )
+            )
+        except StopIteration:
+            raise ValueError(f"Received an invalid mode: {mode}") from None
+
+        await self.info.node.async_set_value(self._current_mode, new_state)
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set new target humidity."""
+        await self.info.node.async_set_value(self._setpoint_value, humidity)
+
+    @property
+    def min_humidity(self) -> int:
+        """Return the minimum humidity."""
+        min_value = DEFAULT_MIN_HUMIDITY
+        try:
+            if self._setpoint_value and self._setpoint_value.metadata.min:
+                min_value = self._setpoint_value.metadata.min
+        # In case of any error, we fallback to the default
+        except (IndexError, ValueError, TypeError):
+            pass
+
+        return min_value
+
+    @property
+    def max_humidity(self) -> int:
+        """Return the maximum humidity."""
+        max_value = DEFAULT_MAX_HUMIDITY
+        try:
+            if self._setpoint_value and self._setpoint_value.metadata.max:
+                max_value = self._setpoint_value.metadata.max
+        # In case of any error, we fallback to the default
+        except (IndexError, ValueError, TypeError):
+            pass
+
+        return max_value
+
+    @property
+    def operating_state(self) -> str | None:
+        """Return the current humidity control operating state."""
+        if self._operating_state:
+            return cast(
+                str,
+                self._operating_state.metadata.states[str(self._operating_state.value)],
+            )
+        return None
+
+    @property
+    def extra_state_attributes(self) -> dict[str, str] | None:
+        """Return the optional state attributes."""
+        attrs = {}
+
+        if state := self.operating_state:
+            attrs[ATTR_OPERATING_STATE] = state
```

Comment:
I updated this PR to use `extra_state_attributes` for this `operating_state` attribute to avoid blocking on a decision for https://github.com/home-assistant/core/pull/65855

Reply:
I suggest we leave this out of the implementation for now. It's better to wait until we know how we want to express this generally.

Reply:
done